src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -636,6 +636,17 @@ int RGWRados::get_required_alignment(const DoutPrefixProvider *dpp, const rgw_po
     return r;
   }
 
+  bufferlist inbl;
+  librados::Rados *rados = get_rados_handle();
+  r = rados->mon_command(
```

Comment:
`RGWRados::get_required_alignment()` is probably being called for each object write, so we don't want to send a mon command every time

`rgw_init_ioctx()` in driver/rados/rgw_tools.cc is where pool creation happens, so we should move this there. it already sends mon commands for `pg_autoscale_bias` and `recovery_priority` for "mostly omap" pools. it would probably make sense to add a new `bool bulk_data` argument to enable this new behavior

then we'd need to thread this option through `open_pool_ctx()` so `RGWRados::get_required_alignment()` can pass `bulk_data=true`

Reply:
> `rgw_init_ioctx()` in driver/rados/rgw_tools.cc is where pool creation happens, so we should move this there. it already sends mon commands for `pg_autoscale_bias` and `recovery_priority` for "mostly omap" pools. it would probably make sense to add a new `bool bulk_data` argument to enable this new behavior
> 
> then we'd need to thread this option through `open_pool_ctx()` so `RGWRados::get_required_alignment()` can pass `bulk_data=true`

ack, thank you very much - changed